### ISSUE TITLE
Segfault in python

### ISSUE THREAD

**Issue by Nonisiuniis** [jruizgit/rules#96]
Few months ago I've experimented with durable_rules (tried to remove threading dependency and convert it to async) and encountered segfaults on some actions. Tested it today with both latest pip version and fresh git pull, still the same result:
```
$ pip install durable_rules
Collecting durable_rules
  Downloading durable_rules-0.33.111.tar.gz ....
[...]

$ python
Python 3.5.3 (default, May  5 2017, 10:33:46)
[GCC 4.9.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import rules
>>> rules.delete_client(1)
Segmentation fault
```

**Comment by jruizgit** [comment]
Hi, thanks for posting the question. Can you please share the code for `rules.delete_client`? It is not a durable_rules class.  A stack trace is also helpful.

**Comment by Nonisiuniis** [comment]
There is no actual code. I was just stepping through the code and tried to understand, how it connects together, if some internals can be queried (number of clients, their state, etc.) and so on. This particular error was triggered when calling close() method on engine.Queue object more than once

https://github.com/jruizgit/rules/blob/68ed6482d2cdb99eac0e3d60597913666c863ac9/libpy/durable/engine.py#L1135

**Comment by jruizgit** [comment]
Hi, I have uploaded a fix, the close method in the queue class can be called more than once. Once the queue has been closed, calling other methods in the class will result in an exception. Please use version 0.33.112.

**Comment by Nonisiuniis** [comment]
Nice shortcut ;-) Is proper fix hard to do? I can't write in C -yet -, so I can just follow the code and see where it ends. From what I see there is _deleteClient_ called in _rete.c_ without checking if the handle actually exists. 

Exactly the same happens with _deleteRuleset_ (I knew there was more than one place that segfaulted). As _deleteRuleset_ is directly mapped to python function _delete_ruleset_, it can be triggered the same way:
```
$ python
Python 3.5.3 (default, May  5 2017, 10:33:46)
[GCC 4.9.4] on linux
>>> import rules
>>> rules.delete_ruleset(1)
Segmentation fault
$
```
